### PR TITLE
Update write_text to include newline variable

### PR DESCRIPTION
**Pull request by pricemg**
As of python3.10 `write_text` in the `pathlib` module also has the optional argument `newline` ([code](https://github.com/python/cpython/blob/fc382d3dd08709a9dc5000a691d3a74f7b4a99ac/Lib/pathlib.py#L1146)). 

Adding in the functionality to the `open` call.

### CODE REVIEW

File: cloudpathlib/cloudpath.py [hunk 1]
```diff
@@ -626,15 +627,20 @@ def write_text(
         data: str,
         encoding: Optional[str] = None,
         errors: Optional[str] = None,
+        newline: Optional[str] = None,
     ) -> int:
         """Open the file in text mode, write to it, and close the file.
 
         NOTE: vendored from pathlib since we override open
         https://github.com/python/cpython/blob/3.8/Lib/pathlib.py#L1244-L1252
```
Comment (pricemg) [inline cloudpathlib/cloudpath.py:635]:
```suggestion
```

File: cloudpathlib/cloudpath.py [hunk 2]
```diff
@@ -626,15 +627,19 @@ def write_text(
...
-        https://github.com/python/cpython/blob/3.8/Lib/pathlib.py#L1244-L1252
+        https://github.com/python/cpython/blob/3.10/Lib/pathlib.py#L1146-L1155
         """
         if not isinstance(data, str):
             raise TypeError("data must be str, not %s" % data.__class__.__name__)
-        with self.open(mode="w", encoding=encoding, errors=errors) as f:
+
+        encoding = io.text_encoding(encoding)
+
+        with self.open(mode='w', encoding=encoding, errors=errors, newline=newline) as f:
```
Comment (pricemg) [inline cloudpathlib/cloudpath.py:642]:
```suggestion
        with self.open(mode="w", encoding=encoding, errors=errors, newline=newline) as f:
```

**Comment by pjbull** [comment]
Looks good to me, thanks @pricemg! Could you also add a note to `HISTORY.md`? After that, I'll merge to a local branch to run the live tests and then into the main branch.

**Comment by pricemg** [comment]
Thanks @pjbull done that now, thought i had more things to look into for those failing CICD checks so happy to hand it over :D

**Comment by pjbull** [comment]
Merging for live test run

**Comment by pjbull** [comment]
Thanks @pricemg, this has now shipped in version `0.16.0`

**Pull request merged**